woocommerce/compatibility/class-sv-wc-order-compatibility.php

```diff
@@ -718,6 +718,42 @@ public static function delete_order_meta( $order, string $meta_key, $meta_value
 	}
 
 
+	/**
+	 * Get global $post or $theorder order object according to HPOS availability.
+	 *
+	 * @since x.y.z
+	 * @return \WC_Order|\WP_Post  object
+	 */
+	public static function get_post_or_order() {
+
+		global $post, $theorder;
+
+		if ( ! is_null( $theorder ) && static::is_order( $theorder ) ) {
+			return $theorder;
+		} else if( ! is_null( $post ) ) {
+			return $post;
+		}
+	
+	}
```

Comment:
I don't think this method would be needed. As it is currently it also returns void / does not handle scenarios where global is neither order or post. But besides, I don't think we want methods that return different objects that aren't related.